outscale/data_source_outscale_flexible_gpu.go

```diff
@@ -58,22 +57,11 @@ func dataSourceOutscaleOAPIFlexibleGpu() *schema.Resource {
 func dataSourceOutscaleOAPIFlexibleGpuRead(d *schema.ResourceData, meta interface{}) error {
 
 	conn := meta.(*OutscaleClient).OSCAPI
-
-	filters, filtersOk := d.GetOk("filter")
-	flexID, IDOk := d.GetOk("flexible_gpu_id")
-
-	if !filtersOk && !IDOk {
-		return fmt.Errorf("One of filters, or flexible_gpu_id must be assigned")
-	}
-
 	req := oscgo.ReadFlexibleGpusRequest{}
-
-	req.Filters = &oscgo.FiltersFlexibleGpu{
-		FlexibleGpuIds: &[]string{flexID.(string)},
+	if filters, filtersOk := d.GetOk("filter"); filtersOk {
+		req.SetFilters(buildOutscaleOAPIDataSourceFlexibleGpuFilters(filters.(*schema.Set)))
 	}
```

Comment:
This is a singular data_source, therefore to avoid matching with multiple data_source user should add
```
filtersilters, filtersOk := d.GetOk("filter")
if !filtersOk {
   return fmt.Errorf("filters must be assigned")
}
```